profiling-ffi/src/profiles.rs

```diff
@@ -434,10 +453,8 @@ pub struct EncodedProfile {
 /// call this twice on the same object);
 #[no_mangle]
 pub unsafe extern "C" fn ddog_prof_EncodedProfile_drop(profile: Option<&mut EncodedProfile>) {
```

Comment:
Unfortunately this does not seem to be correct. I was trying to run the Ruby profiler test suite and I get a crash whenever this function gets called. I was able to reproduce the same issue in the `exporter.cpp` example:

```
$ DD_API_KEY=abc ./exporter foo
free(): invalid pointer
Aborted
```

Reply:
(If I otherwise comment out the drop I can run the Ruby profiler test suite fine with this PR)

Reply:
Yes, I messed up the `ddog_prof_EncodedProfile_drop` when refactoring. Leaving a comment to help me remember.